### ISSUE TITLE
2.x: WithLatestFrom not calling onSubscribe() on `actual` observer

### ISSUE THREAD

**Issue by ZacSweers** [ReactiveX/RxJava#4969]
We're seeing an NPE from the `s` field of `SerializedObserver` not being set when used with `.withLatestFrom()`. Digging around in the source, it looks like it internally is not calling `onSubscribe()` to the `actual` observer field.

Example trace looks like this:

```
java.lang.NullPointerException
	at io.reactivex.observers.SerializedObserver.onNext(SerializedObserver.java:91)
	at io.reactivex.internal.operators.observable.ObservableWithLatestFrom$WithLatestFromObserver.onNext(ObservableWithLatestFrom.java:101)
	at io.reactivex.internal.operators.observable.ObservableFilter$FilterObserver.onNext(ObservableFilter.java:51)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:63)
	at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext(PublishSubject.java:263)
	at io.reactivex.subjects.PublishSubject.onNext(PublishSubject.java:182)
```

The snippet in `SerializedObserver` that corresponds to

```java
    @Override
    public void onNext(T t) {
        if (done) {
            return;
        }
        if (t == null) {
            s.dispose();  // <----- this line, "s" is null because onSubscribe was never called
            onError(new NullPointerException("onNext called with null. Null values are generally not allowed in 2.x operators and sources."));
            return;
        }
```

**Comment by ZacSweers** [comment]
Specifically, the static `WithLatestFromObserver` class in `ObservableWithLatestFrom` never calls `OnSubscribe` on its `actual` before sending other events to it.

**Comment by akarnokd** [comment]
The main problem is that `withLatestFrom` doesn't check for null returned by the `combiner`. I'll post a PR shortly.

**Comment by ZacSweers** [comment]
:+1: thanks!

**Comment by ZacSweers** [comment]
Fixed in #4970
